src/java.desktop/share/classes/javax/swing/JSplitPane.java

```diff
@@ -362,6 +363,32 @@ public JSplitPane(int newOrientation,
 
     }
 
+    /**
+     * {@inheritDoc}
+     * @param orientation {@inheritDoc}
+     */
+    @Override
+    public void setComponentOrientation(ComponentOrientation orientation) {
+        super.setComponentOrientation(orientation);
+        Component leftComponent = this.getLeftComponent();
+        Component rightComponent = this.getRightComponent();
+        if (this.getComponentOrientation().equals(ComponentOrientation.RIGHT_TO_LEFT)) {
+            if (rightComponent != null) {
```

Comment:
I hope `null` is redundant here, because it is handled inside `setRightComponent/setLeftComponent`. I don't see any difference with and without `null` check while testing in my local. Any particular reason for adding `null` check here?

Reply:
Probably, but I kept it as it is similar to the constructor where it does a null check..